lib/egg.js

```diff
@@ -203,10 +204,12 @@ class EggCore extends KoaApplication {
   /**
    * Execute scope after loaded and before app start
    *
-   * @param  {Function|GeneratorFunction|AsyncFunction} scope function will execute before app start
+   * @param  {Function|GeneratorFunction|AsyncFunction} fn - The function to be executed, it
+   * can be a function, generator function or an async function.
    */
-  beforeStart(scope) {
-    this.lifecycle.registerBeforeStart(scope);
+  beforeStart(fn) {
+    assert(is.function(fn), 'beforeStart needs a function.');
```

Comment:
In lifecycle, beforeStart and beforeClose are deprecated, so this shouldn't be changed.

Reply:
@popomore：No. Your validation rule just exists in the 'lifecycle.js', I just remove them out of there and put it here ealier.

Reply:
I just think you don't need to put validation rules so deep in the lifecycle, maybe your 'app.use' method, to check validation rules first before going into the lifecycle is a better choice, so this is just a refactor INSTEAD OF changing anything else.